backend/src/v5/processors/teamspaces/audits.js

```diff
@@ -0,0 +1,68 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { createResponseCode, templates } = require('../../utils/responseCodes');
+const { PassThrough } = require('stream');
+const archiver = require('archiver');
+archiver.registerFormat('zip-encrypted', require('archiver-zip-encrypted'));
+const { templates: emailTemplates } = require('../../services/mailer/mailer.constants');
+const { generateHashString } = require('../../utils/helper/strings');
+const { getActionLog } = require('../../models/teamspaces.audits');
+const { getUserByUsername } = require('../../models/users');
+const { sendEmail } = require('../../services/mailer');
+
+const Audit = {};
+
+const createAuditLogArchive = async (actions) => {
+	try {
+		const password = generateHashString();
+		const stream = new PassThrough();
+
+		const archive = archiver.create('zip-encrypted', {
+			zlib: { level: 1 },
+			encryptionMethod: 'aes256',
+			password,
+		});
+
+		const content = `[${actions.map((action) => JSON.stringify(action)).join(',')}]`;
+		archive.append(content, { name: 'audit.json' });
+
+		const archiveReady = new Promise((resolve, reject) => {
+			archive.on('end', resolve);
+			archive.on('error', reject(templates.streamError));
```

Comment:
this syntax looks weird, the on function should be expecting a function, you're passing a function call instead of the function itself?

We're also not keeping track of what is the error which will make debugging very difficult